src/main/scala/Question7.scala

```diff
@@ -0,0 +1,9 @@
+package questions.question7
+
+def winner(votes: List[String]): Option[String] = {
+  votes.groupBy(x => x).toList.sortBy(_._2.size).reverse match {
+    case x :: Nil => None
+    case x :: y :: xs => if (x._2.size == y._2.size || x._2.size.toDouble/votes.size.toDouble<=0.5) None else Some(x._1)
+    case _ => None
+  }
+}
```

Comment:
I can't argue with that, it was a tad bit rushed and then corrected on the metro! At least I know what my next question is going to be now though :-)